File: CS/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraRichEdit;

namespace RichEditTwoControlsOneToolbar {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void richEditControl_Enter(object sender, EventArgs e) {
            RichEditControl richEditControl = (RichEditControl)sender;
            richEditBarController1.RichEditControl = richEditControl;
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit

Namespace RichEditTwoControlsOneToolbar
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub richEditControl_Enter(ByVal sender As Object, ByVal e As EventArgs) Handles richEditControl1.Enter, richEditControl2.Enter
			Dim richEditControl As RichEditControl = CType(sender, RichEditControl)
			richEditBarController1.RichEditControl = richEditControl
		End Sub
	End Class
End Namespace